From da3633c2e6200dba97b6299bd9b5ee19de948d3e Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Wed, 14 Nov 2018 22:32:05 -0800
Subject: [PATCH] Fix finding ld.so.conf on Clear Linux

Clear Linux has a "stateless" /etc, meaning that all files there are
supposed to belong to the user. The ld.so.cache file is instead in
/var/cache/ldconfig.
---
 client/icecc-create-env.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/client/icecc-create-env.in b/client/icecc-create-env.in
index b862636..7255135 100755
--- a/client/icecc-create-env.in
+++ b/client/icecc-create-env.in
@@ -476,7 +476,11 @@ done
 if test -x /sbin/ldconfig -a "$is_linux" = 1; then
    mkdir -p $tempdir/var/cache/ldconfig
    /sbin/ldconfig -r $tempdir
-   new_target_files="$new_target_files etc/ld.so.cache"
+   for candidate in etc var/cache/ldconfig; do
+        test -e $tempdir/$candidate/ld.so.cache || continue;
+        new_target_files="$new_target_files $candidate/ld.so.cache"
+        break
+   done
 fi
 
 md5sum=NONE
-- 
2.19.1

